eng/pipelines/templates/steps/az-login.yml

```
@@ -15,5 +15,10 @@ steps:
 
       az account set `
         --subscription "$($subscriptionConfiguration.SubscriptionId)"
+
+      echo "##vso[task.setvariable variable=ARM_CLIENT_ID;issecret=true]$($subscriptionConfiguration.TestApplicationId)"
+      echo "##vso[task.setvariable variable=ARM_CLIENT_SECRET;issecret=true]$($subscriptionConfiguration.TestApplicationSecret)"
+      echo "##vso[task.setvariable variable=ARM_TENANT_ID;issecret=true]$($subscriptionConfiguration.TenantId)"
```

* Variables containing secrets use `kebab-casing` to match secret names in the rest of our engineering system. 
* Use `Write-Host` 

```suggestion 
            # Export service principal auth information for terraform testing
            Write-Host "##vso[task.setvariable variable=arm-client-id;issecret=true]$($subscriptionConfiguration.TestApplicationId)"
      Write-Host "##vso[task.setvariable variable=arm-client-secret;issecret=true]$($subscriptionConfiguration.TestApplicationSecret)"
      Write-Host "##vso[task.setvariable variable=arm-tenant-id;issecret=true]$($subscriptionConfiguration.TenantId)"
```